libs/platform/src/lib/components/list/list.component.ts

```diff
@@ -0,0 +1,631 @@
+import {
+    ChangeDetectionStrategy, Component, Input, ViewEncapsulation,
+    ContentChildren, QueryList, HostBinding, ViewChild,
+    ElementRef, AfterContentInit, Output, EventEmitter,
+    HostListener, ChangeDetectorRef, OnInit, AfterViewInit, ContentChild, Self, Optional, SkipSelf, Host, OnDestroy
+} from '@angular/core';
+import { FocusKeyManager } from '@angular/cdk/a11y';
+import { UP_ARROW, DOWN_ARROW, ENTER, SPACE } from '@angular/cdk/keycodes';
+import { NgControl, NgForm } from '@angular/forms';
+import { SelectionModel } from '@angular/cdk/collections';
+import { Subscription, Subject, of } from 'rxjs';
+import { takeUntil, delay, tap } from 'rxjs/operators';
+import { BaseComponent } from '../base';
+import { BaseInput } from '../form/base.input';
+import { ListDataSource, isDataSource } from '../../domain/data-source';
+import { ContentDensity, FormFieldControl } from '../../components/form/form-control';
+import { BaseListItem, ListItemDef } from './base-list-item';
+import { ListConfig } from './list.config';
+import { FormField } from '../form/form-field';
+
+
+export type SelectionType = '' | 'multi' | 'single' | 'delete';
+export type ListType = 'inactive' | 'active' | 'detail';
+let nextListId = 0;
+let nextListGrpHeaderId = 0;
+export type FdpListDataSource<T> = ListDataSource<T> | T[];
+export class SelectionChangeEvent {
+    selectedItems: BaseListItem[];
+    index: number;
+}
+
+/**
+ * The List component represents a container for list item types.
+ * It is used to display a list features.
+ */
+@Component({
+    selector: 'fdp-list',
+    templateUrl: './list.component.html',
+    encapsulation: ViewEncapsulation.None,
+    styleUrls: ['./list.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [{ provide: FormFieldControl, useExisting: ListComponent, multi: true }]
+})
+
+
+export class ListComponent extends BaseInput implements OnInit, AfterViewInit, AfterContentInit, OnDestroy {
```

Comment:
Please consider using CollectionBasedINput, There is existing PR for ComboBox  - it should use the same approach

Reply:
updated